packages/cli/src/commands/generate.ts

```diff
@@ -0,0 +1,69 @@
+import { Command, Option } from 'clipanion';
+import { cp, readFile, writeFile, rename } from 'node:fs/promises';
+import { dirname, join } from 'node:path';
+import { spawnSync } from 'node:child_process';
+import { pascalCase } from 'change-case';
+import { BaseCommand } from './base';
+
+export class GenerateCommand extends BaseCommand {
+  static override paths = [['generate'], ['gen']];
+
+  static override usage = Command.Usage({
+    // category: `My category`,
+    description: `Generate a boilerplated project for a custom \`ConfigStore\``,
+    // details: `
+    //   A longer description of the command with some \`markdown code\`.
+
+    //   Multiple paragraphs are allowed. Clipanion will take care of both reindenting the content and wrapping the paragraphs as needed.
+    // `,
+    // examples: [
+    //   [`A basic example`, `$0 my-command`],
+    //   [`A second example`, `$0 my-command --with-parameter`],
+    // ],
+  });
+
+  directory = Option.String('--dir,--di', {
+    description: `Location to generate the \`ConfigStore\` project`,
+    required: true,
+  });
+
+  name = Option.String('--name,--na', {
+    description: `Name of the \`ConfigStore\``,
+    required: true,
+  });
+
+  async execute() {
+    await this.init();
+
+    console.log(`Copying from ${join(dirname(import.meta.dirname), 'integration_template')} to ${this.directory}`);
+    console.log('Generating project...');
+    await this.generateBoilerplate();
+    await this.overrideTemplates();
+    await this.installDependencies();
+  }
+
+  private async generateBoilerplate() {
+    console.log(`Copying project boilerplate...\r`);
+    await cp(join(dirname(import.meta.dirname), 'integration_template'), this.directory, { recursive: true });
```

Comment:
If we go down the `write-from-local` route rather than downloading the boilerplate, we'll probably want to bring all the contents into strings in JS files rather than `cp`-ing them - since we want the command line tool to bundle to a single executable.

Reply:
we can potentially tight between the boilerplate dir / repo to the cli deployment in a way that they will be bundled as artifacts with the tsup of the cli / or use digit as a dev dependency and trigger it as part of the cli release process to download and bundled them.